addons/mail/models/discuss/mail_guest.py

```diff
@@ -199,6 +199,8 @@ def _find_or_create_for_channel(self, channel, name, country_id, timezone, post_
                     "timezone": timezone,
                 }
             ).sudo(False)
+        if self.env.user._is_public():
+            channel = channel.with_context(guest=guest)
```

Comment:
When called from `action_send_chat_request` it is not correct to update the context.

Reply:
I guess that's why there is `if self.env.user._is_public():` but this is not great.

Reply:
In this case the user is not public, so the context is not updated.

Reply:
Can't we just make sure `add_members` sends the right stuff?

Reply:
Or we stop using that method in `action_send_chat_request`?

Reply:
It makes no sense to check guest from context in that case anyway. It will actually bug if the current user has an old guest cookie.